akka-docs/src/main/paradox/persistence-typed.md

```diff
@@ -1,32 +1,179 @@
+  change without warning or deprecation period and it is not recommended to use
+  this module in production just yet—you have been warned.
+  
+This module only has a Scala DSL. See [#24193](https://github.com/akka/akka/issues/24193) 
+to track progress and to contribute to the Java DSL.
+  
+@@@
+
+To use typed persistence add the following dependency:
+
+@@dependency [sbt,Maven,Gradle] {
+  group=com.typesafe.akka
+  artifact=akka-persistence-typed_2.11
+  version=$akka.version$
+}
+
+
+
+Persistence is a library for building event sourced actors. For background about how it works
+see the @ref:[untyped docs section](persistence.md). This documentation shows how the typed API for persistence
+works and assumes you know what is meant by `Command`, `Event` and `State`.
+
+Let's start with a simple example. The minimum required for a `PersistentActor` is:
+
+Scala
+:  @@snip [BasicPersistentActorSpec.scala]($akka$/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentActorSpec.scala) { #structure }
+
+The first important thing to notice is the `Behavior` of a `PersistentActor` is typed to the type of the `Command` 
+because this type of message a persistent actor should receive. In akka typed this is now enforced by the type system.
+The event and state are only used internally.
+
+The parameters to `PersistentActor.immutable` are::
+
+* `persistenceId` is the unique identifier for the persistent actor.
+* `initialState` defines the `State` when the entity is first created.
+* `commandHandler` defines how to handle command and optional functions for other signals, e.g. `Termination` messages if `watch` is used.
+* `eventHandler` updates the current state when an event has been persisted.
+
+### Command handler
+
+The command handler is a function with 3 parameters for the `ActorContext`, current `State`, and `Command`.
+
+A command handler returns an `Effect` directive that defines what event or events, if any, to persist.
+
+* `Effect.persist` will persist one single event or several events atomically, i.e. all events
+  are stored or none of them are stored if there is an error
+* `Effect.none` no events are to be persisted, for example a read-only command
+* `Effect.unhandled` the command is unhandled (not supported) in current state
+
+External side effects can be performed after successful persist with the `andThen` function. 
+In the example below a reply is sent to the `replyTo`. Note that the new state after applying 
+the event is passed as parameter to the `andThen` function.
+
+### Event handler
+
+When an event has been persisted successfully the current state is updated by applying the 
+event to the current state with the `eventHandler` function. 
+
+The event handler returns the new state, which must be immutable so you return a new instance of the state. 
+The same event handler is also used when the entity is started up to recover its state from the stored events.
```

Comment:
Add info on "It is not recommended to perform side effects in the event handler, as those are also executed during recovery of an persistent actor"

Reply:
+2